pkg/apis/ceph.rook.io/v1/resources.go

```diff
@@ -60,11 +66,45 @@ func GetMonResources(p ResourceSpec) v1.ResourceRequirements {
 	return p[ResourcesKeyMon]
 }
 
-// GetOSDResources returns the placement for the OSDs
+// GetOSDResources returns the placement for the all OSDs
 func GetOSDResources(p ResourceSpec) v1.ResourceRequirements {
 	return p[ResourcesKeyOSD]
 }
 
+// GetOSDResourceKeyForDeviceClass returns key name for device class in resources spec
+func GetOSDResourceKeyForDeviceClass(deviceClass string) string {
+	return ResourcesKeyOSD + "-" + deviceClass
+}
+
+// GetOSDResourcesForDeviceClass returns the placement for the OSDs of specified device class
+func GetOSDResourcesForDeviceClass(p ResourceSpec, deviceKey string) v1.ResourceRequirements {
+	switch deviceKey {
+	case ResourcesKeyHDDDeviceOSD:
+		return GetOSDWithHDDResources(p)
+	case ResourcesKeySSDDeviceOSD:
+		return GetOSDWithSSDResources(p)
+	case ResourcesKeyNVMEDeviceOSD:
+		return GetOSDWithNVMEResources(p)
+	default:
+		return GetOSDResources(p)
+	}
+}
+
+// GetOSDWithHDDResources returns the placement for the OSDs with HDD devices
+func GetOSDWithHDDResources(p ResourceSpec) v1.ResourceRequirements {
```

Comment:
Can we not define these specific consts and functions for hdd, ssd, and nvme? In other places I see we construct the name with the pattern `osd-<type>`. Can we not just have a general function for this as well?